comment/comment.go

```diff
@@ -59,6 +61,25 @@ func (m *GormCommentRepository) Create(ctx context.Context, u *Comment) error {
 	return nil
 }
 
+// Save a single comment
+func (m *GormCommentRepository) Save(ctx context.Context, comment *Comment) (*Comment, error) {
+	c := Comment{}
+	tx := m.db.Where("id=?", comment.ID).First(&c)
+	if tx.RecordNotFound() {
+		// treating this as a not found error: the fact that we're using number internal is implementation detail
+		return nil, errors.NewNotFoundError("comment", comment.ID.String())
+	}
+	if err := tx.Error; err != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	tx = tx.Save(comment)
```

Comment:
We don't have audit trail in other parts of the system now (e.g, work item, work item link etc.), So is this required now or can we do it later?